zilliqa/src/db.rs

```diff
@@ -271,12 +271,15 @@ impl Db {
         Ok(Some(base_path.join("checkpoints").into_boxed_path()))
     }
 
+    /// Fetch checkpoint data from file and initialise db state
+    /// Return checkpointed block and transactions which must be executed after this function
+    /// Return None if checkpoint already loaded
     pub fn load_trusted_checkpoint<P: AsRef<Path>>(
         &self,
         path: P,
         hash: &Hash,
         our_shard_id: u64,
-    ) -> Result<Block> {
+    ) -> Result<Option<(Block, Vec<SignedTransaction>, Block)>> {
         // For now, only support a single version: you want to load the latest checkpoint, anyway.
         const SUPPORTED_VERSION: u32 = 1;
```

Comment:
Shall we bump this to version 2 since we're now including transactions?

I don't care about backwards compatibility, but it might be helpful to ensure we don't try to load an older checkpoint with this new version of the code and interpret some other random bytes as transactions or similar.

Reply:
cool updated